### PR TITLE
de-nosing olympia/amo/ app.  Includes base TestCase classes.

### PR DESCRIPTION
**Pull request by eviljeff**

### CODE REVIEW

File: src/olympia/amo/tests/test_storage_utils.py [hunk 1]
```diff
@@ -33,21 +32,22 @@ def test_storage_walk():
         results = [(dir, set(subdirs), set(files))
                    for dir, subdirs, files in sorted(walk_storage(tmp))]
 
-        yield (eq_, results.pop(0),
```
Comment (eviljeff) [inline src/olympia/amo/tests/test_storage_utils.py]: I couldn't see what difference having all these in a yield was doing (wouldn't pop always remove and return the first entry anyway?)

File: src/olympia/amo/tests/test_amo_utils.py [hunk 2]
```diff
@@ -26,12 +25,14 @@
 u = u'Ελληνικά'
 
 
-def test_slug_validator():
-    eq_(slug_validator(u.lower()), None)
-    eq_(slug_validator('-'.join([u.lower(), u.lower()])), None)
-    assert_raises(ValidationError, slug_validator, '234.add')
-    assert_raises(ValidationError, slug_validator, 'a a a')
-    assert_raises(ValidationError, slug_validator, 'tags/')
+class TestSlugValidator(TestCase):
```
Comment (EnTeQuAk) [inline src/olympia/amo/tests/test_amo_utils.py]: I might be wrong but imho `TestCase` adds database functionality to the test-suite and we don't need that for this test so we could potentially skip inheriting from `TestCase` - we don't have to anyway with pytest. The `assertRaises` could become `pytest.raises(Exception, func, *args, **kwargs)`

but let's not inherit from `TestCase` if we don't have to, 'cause it'll add quite a bit of execution time and our test-suite already is slow as hell.

File: src/olympia/amo/tests/test_storage_utils.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/amo/tests/test_storage_utils.py]: I think the `yield`s were used to generate new test-cases based on what's in `results` but I don't think they added anything useful. Might be a good example of using `pytest.mark.parametrize` imho though.

**Comment by EnTeQuAk** [comment]
You deserve all the :cookie: and :cake: and :beers: in the world, nice work!

r+wc

File: src/olympia/amo/tests/test_amo_utils.py (hunk 2 above)
Comment (eviljeff) [inline src/olympia/amo/tests/test_amo_utils.py]: okay, I'll change to pytest.raises.  

I'm not sure that one function is going to make a difference though when we have many hundreds of subclasses of TestCase - going through and swapping out TestCase for one of the simpler superclasses where database rollback isn't needed would be a solution but I'll leave that for someone else :)

**Comment by diox** [comment]
regarding `TestCase`, I wonder if we could find out through some pytest magic if there are tests that did not need the database at all, and mark them as not requiring db to speed things up...

**Comment by EnTeQuAk** [comment]
Unfortunately pytest-django makes database access available for every class that inherits `django.test.TestCase` which will probably make it quite hard to distinquish where it's being used. We could potentially add some temporary hack to `setUp` / `tearDown` and check wether or not the database is being used (anything in `django.db.connection.queries`) and log that somewhere - it's not a 100% thing but should be good enough.

**Pull request merged**